packages/story-editor/src/utils/getResourceBaseColor.js

```diff
@@ -46,16 +46,19 @@ function getImgNodeKey(elementId) {
   return `${IMG_NODE}_${elementId}`;
 }
 
-export function getMediaBaseColor(resource, onBaseColor) {
+export function getResourceBaseColor(resource) {
   const { type, src, poster } = resource;
-  setOrCreateImage({
-    src: type === 'video' ? poster : src,
+
+  const imageSrc = type === 'image' ? src : poster;
+
+  if (!imageSrc) {
+    return Promise.reject(new Error('No source to image'));
+  }
+  return setOrCreateImage({
+    src: imageSrc,
     width: 10,
     height: 'auto',
```

Comment:
@miina Is such a small image here by design? We could use the height / width from the resource and get the full size image and pick from that.

Reply:
Probably for performance reasons since it's an average color anyway. Would using a full size image make a difference?

Reply:
I am guessing, that a 10x10 image compared to a 600x600 image would get a different primary color from colorthief? Don't you agree?

Reply:
Not sure, would need testing out to see if it makes a difference that could actually result in an incorrect color that would not fulfil its purpose.

It's possible to test it here for example (in the Try it yourself section): https://lokeshdhakar.com/projects/color-thief/